cpp/include/cudf/strings/string.cuh

```diff
@@ -52,6 +52,31 @@ inline __device__ bool is_integer(string_view const& d_str)
            thrust::seq, begin, end, [] __device__(auto chr) { return chr >= '0' && chr <= '9'; });
 }
 
+/**
+ * @brief Returns true if input contains the not-a-number string.
+ *
+ * The following are valid for this function: "NAN" and "NaN"
+ * @param d_str input string
+ * @return true if input is as valid NaN string.
+ */
+inline __device__ bool is_nan_str(string_view const& d_str)
+{
+  return d_str.compare("NAN", 3) == 0 || d_str.compare("NaN", 3) == 0;
```

Comment:
From https://en.cppreference.com/w/cpp/string/basic_string/stof

>not-a-number expression. It consists of the following parts:
> - (optional) plus or minus sign
> - NAN or NAN(char_sequence) **ignoring case** of the NAN part. char_sequence can only contain digits, Latin letters, and underscores. The result is a quiet NaN floating-point value.

Wouldn't this mean that all possible combinations of upper/lower case 'N', 'A', 'N' characters would be valid?

Reply:
Oh wow. I missed that part of the text. 
I wrote a quick program and verified that `std::stof` does indeed ignore case for `nan` and `inf` strings. 
That should not be difficult to support for this.